### DISCUSSION TITLE
How To call this inside event callback ?

### DISCUSSION THREAD

**Discussion by bagoesz21** [inertiajs/inertia#503] [Help (Vue)] [answered]
Hi, i have inertia post. When event success called, i want call my function inside event callback. Here look :

```
Inertia.post('/users', data, {
  onBefore: (visit) => {},
  onStart: (visit) => {},
  onProgress: (progress) => {},
  onSuccess: (page) => {
    this.myFunction();
 },
  onError: (errors) => {},
  onCancel: () => {},
  onFinish: () => {},
})
```
But, i'm got error, myFunction undefined.
How, i call my function inside event callback ? Thanks ..

**Comment by claudiodekker** [comment, marked as answer]
Hi,

This almost guaranteed has to do with the fact [that the behaviour of `this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) inside of an arrow function differs from the way a normal function is defined: 

- In a [regular function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function), the [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) variable refers to the object in which that function exists.
- In an [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions), the [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) variable refers to the [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) value in the parent scope.

Since you are already using the [arrow function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) syntax for your `onSuccess` callback, [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) is referring to the scope in which your `Inertia.post` was call is defined. It's likely that that scope, or one of the scopes surrounding that one, is a [regular function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function), which is causing `this` to be 'rewritten', therefore losing the context that allows you to access `myFunction`.

Hope this helps! If not, it would be appreciated if you could update your post with more relevant code of the context in which these functions exist, as the code you provided seems to be correct!

**Reply by bagoesz21** [reply]
[https://inertiajs.com/manual-visits#event-callbacks](https://inertiajs.com/manual-visits#event-callbacks)
Okey, i got this solution. Thanks
